src/clojure/com/interrupt/edgar/core/analysis/lagging.clj

```diff
@@ -105,23 +13,40 @@
    :etal - other keys to emit in each result map
 
    ** This function assumes the latest tick is on the right**"
-  [options tick-list]
-  {:pre [(time-increases-left-to-right? tick-list)]}
-
-  (let [{input-key :input
-         output-key :output
-         etal-keys :etal
-         :or {input-key :last-trade-price
-              output-key :last-trade-price-average
-              etal-keys [:last-trade-price :last-trade-time :uuid]}} options]
-
-    (as-> tick-list v
-      (sum v input-key)
-      (average v (count tick-list))
-      (-> etal-keys
-          (zipmap (map #(% (last tick-list)) etal-keys))
-          (assoc output-key v
-                 :population tick-list)))))
+  [options ticks]
+  {:pre [(not-empty ticks)
+         (time-increases-left-to-right? ticks)]}
+  (let [{:keys [input output etal]
+         :or {input :last-trade-price
+              output :last-trade-price-average
+              etal [:last-trade-price :last-trade-time :uuid]}} options
+        xs (map input ticks)]
+    (-> (last ticks)
+        (select-keys etal)
+        (assoc output (math/mean xs) :population ticks))))
```

Comment:
Also, since `simple-moving-average` has changed, can you add a test, and try streaming it to your browser?